lib/matplotlib/projections/polar.py

```diff
@@ -987,12 +987,26 @@ def set_thetalim(self, *args, **kwargs):
         where minval and maxval are the minimum and maximum limits. Values are
         wrapped in to the range :math:`[0, 2\pi]` (in radians), so for example
         it is possible to do ``set_thetalim(-np.pi / 2, np.pi / 2)`` to have
-        an axes symmetric around 0.
+        an axes symmetric around 0. If not within range :math:`[0, 2\pi]` a
+        ValueError is raised.
         """
+        thetamin = 0
+        thetamax = 0
         if 'thetamin' in kwargs:
-            kwargs['xmin'] = np.deg2rad(kwargs.pop('thetamin'))
+            thetamin = np.deg2rad(kwargs.pop('thetamin'))
```

Comment:
Is all this argument dance because we have to check the generic `*args, *kwargs` signature? If so, we should parse to that `thetamax` and `thetamin`, check that once and pass it to `set_xlim()` explicitly.

Later on, we should make the method signature explicit, which will simplify the check. Ideally, one would to this before adding the check, but since that will involve an API-change with a deprecation period, we would not be able to add the simplified check for the next to minor releases. So the more complex check is necessary for now.

Reply:
Yes, the if statements there are to check the possible signatures. The signatures allowed for set_thetalim are as follows:
1) set_thetalim(minval, maxval): Set the limits in radians.
2) set_thetalim(thetamin=minval, thetamax=maxval): Set the limits in degrees.
These were taken straight out of the documentation for function set_thetalim.

By "If, so we should parse to that thetamax and thetamin, check that once and pass it to set_xlim() explicitly" do you mean signature 1 ( radians ) would not be used anymore?

Reply:
Ultimately (unless we decide to change the awkward rad/degree API altogether) we could at least do `def set_thetalim(min_rad, max_rad, /, *, thetamin, thetamax)`, which is more explicit but has the exact same semantics in terms of positional and keyword arguements. But that would have to wait until the min. supported Python version is 3.8, which introduced the positional-only arguments.

Reply:
for this change I have made changes. Made an explicit call to set_xlim using parameter thetamin, thetamax, left and right. Please let me know if further changes need to be made. I do agree making the set_thetalim explicit in the future when 3.8 rolls out.

Reply:
Passing the parameters on consistently and checking them on the go is surprisingly complex. This code covers the documented API versions using two kwargs or two positional args. It does not catch all edge cases like `set_thetalim(0, thetamax=180` which technically works as well.

I'll think about this but I think that's ok.

I'd move these theta parsing down so that you have the order
- args parsing
- args checks
- theta parsing
- theta checks